ui/src/modal.ts

```diff
@@ -1,16 +1,25 @@
 import { derived, get, writable } from "svelte/store";
+import type { SvelteComponent } from "svelte";
+import Blank from "../Screen/Blank.svelte";
 
 type OnHide = () => void;
 // eslint-disable-next-line @typescript-eslint/no-empty-function
 const doNothing = () => {};
 
 type ModalOverlay =
-  | { show: true; route: string; onHide: OnHide }
-  | { show: false };
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | { show: true; modal: typeof SvelteComponent; onHide: OnHide; props: any }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | { show: false; modal: typeof SvelteComponent; onHide: OnHide; props: any };
 
-const overlayStore = writable<ModalOverlay>({
+const hidden = {
   show: false,
-});
+  modal: Blank,
+  onHide: doNothing,
+  props: {},
+};
+
+const overlayStore = writable<ModalOverlay>(hidden);
```

Comment:
This could be 
```suggestion
const overlayStore = writable<ModalOverlay | null>(null);
```
to avoid the artificial blank component

Reply:
27437a6fc0f812de049d97cc87da14bea62ea969
975817fc65d6d1250a151144d29c10285b329d29